### PR TITLE
[bug](not in) if not in (null) could eos early

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
ref: https://github.com/apache/doris/pull/15466
not in(any, null) == (x!=any and x!=null)
(x!=null) = null = false, so not in(any, null) return always false

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/exec/scan/vscan_node.cpp [hunk 1]
```diff
@@ -864,7 +864,8 @@ Status VScanNode::_normalize_not_in_and_not_eq_predicate(VExpr* expr, VExprConte
 
         HybridSetBase::IteratorBase* iter = state->hybrid_set->begin();
         auto fn_name = std::string("");
-        if (!is_fixed_range && state->hybrid_set->contain_null()) {
+        // https://github.com/apache/doris/pull/15466
+        if (state->hybrid_set->contain_null()) {
```
Comment (yiguolei) [inline be/src/vec/exec/scan/vscan_node.cpp:868]: Any  regression tests?

**Comment by zhangstar333** [comment]
run buildall

**Comment by xiaokang** [comment]
pls add testcase

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**